ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/AbstractPartitionPlan.java

```diff
@@ -295,9 +295,9 @@ public List<SplitTask> splitTasks(int targetTaskCount) {
       // collect
       List<SplitTask> results = Lists.newArrayListWithCapacity(packed.size());
       for (List<FileTask> fileTasks : packed) {
-        Set<IcebergDataFile> rewriteDataFiles = Sets.newHashSet();
-        Set<IcebergDataFile> rewritePosDataFiles = Sets.newHashSet();
-        Set<IcebergContentFile<?>> deleteFiles = Sets.newHashSet();
+        Set<DataFile> rewriteDataFiles = Sets.newHashSet();
+        Set<DataFile> rewritePosDataFiles = Sets.newHashSet();
+        Set<ContentFile<?>> deleteFiles = Sets.newHashSet();
 
```

Comment:
That's true, but I don't think there will be duplicate `DateFile` or `DeleteFIle` with the same path.

Reply:
> That's true, but I don't think there will be duplicate `DateFile` or `DeleteFile` with the same path.

Will there be different objects under the same path, such as `datafile` objects generated by two different plantasks

Reply:
1.A notable example is that a delete file will be associated with multiple data files, so it will be added multiple times. 2.Another more subtle example is that when planning tasks, a data file with `ContentFile#splitOffsets` may appear in multiple ScanTasks, resulting in duplicate data file appearances.

Reply:
AFAIK, there only will be one `DateFile` or `DeleteFile` object for different manifest entry in iceberg, so the default implementation for `equals` and `hasCode` is okay here.

Reply:
If it's confirmed that multiple appearances of `DataFile` or `DeleteFile` refer to the same object, then I think it's okay. Perhaps I need to take some time to verify it.